### ISSUE TITLE
RDS parameter group keeps being modified, even though I have applied many times

### ISSUE THREAD

**Issue by shinchan79** [hashicorp/terraform-provider-aws#22028]
# Description
I upgrade Terraform from v0.12 to v0.13, and encounter the following:

```
Terraform will perform the following actions:

  # aws_db_parameter_group.rds_parameter_group["rds_instance_001"] will be updated in-place
  ~ resource "aws_db_parameter_group" "rds_parameter_group" {
        id          = "my-web-rds-pg"
        name        = "my-web-rds-pg"
        tags        = {}
        # (4 unchanged attributes hidden)

      + parameter {
          + apply_method = "immediate"
          + name         = "time_zone"
          + value        = "asia/tokyo"
        }
      - parameter {
          - apply_method = "pending-reboot" -> null
          - name         = "time_zone" -> null
          - value        = "Asia/Tokyo" -> null
        }
        # (7 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```
My Terraform configuration file:

```
resource "aws_db_parameter_group" "rds_parameter_group" {
  for_each = var.rds_instances
  name     = "${each.value.instance_name}-pg"
  family   = each.value.family

  parameter {
    name  = "character_set_client"
    value = "utf8"
  }

  parameter {
    name  = "character_set_connection"
    value = "utf8"
  }

  parameter {
    name  = "character_set_database"
    value = "utf8"
  }

  parameter {
    name  = "character_set_filesystem"
    value = "utf8"
  }

  parameter {
    name  = "character_set_results"
    value = "utf8"
  }

  parameter {
    name  = "character_set_server"
    value = "utf8"
  }

  parameter {
    name  = "log_bin_trust_function_creators"
    value = "1"
  }

  parameter {
    name  = "time_zone"
    value = "asia/tokyo"
  }
}
```

# Expected behavior:

After I ran ``terraform apply``, the changes will disappear, Terraform show message **"No changes. Your infrastructure matches the configuration."**

# Actual behaviour:

After I apply, I run ``terraform plan``  again and RDS parameter group keep being modified like above.

# Steps to Reproduce
1. ``terraform apply``
2. ``terraform plan``

**Comment by andrewpage** [comment]
Having the same problem here.

**Comment by Schnitzel** [comment]
got the same problem, `time_zone` is a `dynamic` setting in AWS:
<img width="1642" alt="Screen Shot 2021-12-29 at 11 26 26" src="https://user-images.githubusercontent.com/750999/147683145-4967814c-5808-4831-a494-c460ea581c51.png">

so this should be `apply_method = "immediate"` but even setting this in terraform directly, somehow the state always thinks that it's `apply_method = "pending-reboot"` which would be a `static` setting and then shows a change in a plan.

This happens to us also with:
- `innodb_lock_wait_timeout` 
- `table_open_cache`
- `table_definition_cache`
- `key_buffer_size`
and many more.

**Comment by Schnitzel** [comment]
also setting `apply_method = "pending-reboot"` on the affected parameters removes the change in the next terraform apply, but I feel this is more a workaround as these parameters should not need `pending-reboot`

**Comment by matthewtckr** [comment]
I'm seeing this with v3.71.0 all the way back to v3.55.0. It does not occur in v3.54.0 or v3.46.0 (the version I was mainly using until trying to update the provider this week).

It looks like it may be caused by #18818, committed and merged starting in v3.55.0, but I'm not very familiar with the provider code.

**Comment by gclough** [comment]
Dupe of #20660 ?

**Comment by jvalente11** [comment]
I believe this issue can be closed now.

See the merged PR #24737 for a demonstration of why this is a behavioral artifact of the AWS API and not Terraform itself.  The PR adds a NOTE to the latest `aws_db_parameter_group` resource [documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_parameter_group), explaining why this is happening.

**Comment by justinretzolk** [comment]
Hey y'all 👋 Based on the details @jvalente11 included in their PR -- which added some information around this to the resource documentation -- this looks to be an upstream issue with AWS. Since the behavior is now documented, we'll close this issue for now, but I'll also be following up with a support ticket to AWS to report the unexpected behavior.
